src/sentry/tasks/store.py

```diff
@@ -193,14 +212,25 @@ def delete_raw_event(project_id, event_id, allow_hint_clear=False):
                 ProjectOption.objects.set_value(project, 'sentry:sent_failed_event_hint', False)
 
 
-def create_failed_event(cache_key, project_id, issues, event_id, start_time=None):
+def create_failed_event(cache_key, project_id, issues, event_id, start_time=None,
+                        reprocessing_rev=None):
     """If processing failed we put the original data from the cache into a
     raw event.  Returns `True` if a failed event was inserted
     """
     reprocessing_active = ProjectOption.objects.get_value(
         project_id, 'sentry:reprocessing_active', REPROCESSING_DEFAULT
     )
 
+    # In case there is reprocessing active but the current reprocessing
+    # revision is already different than when we started, we want to
+    # immediately retry the event.  This resolves the problem when
+    # otherwise a concurrent change of debug symbols might leave a
+    # reprocessing issue stuck in the project forever.
+    if reprocessing_active and \
+       reprocessing.get_reprocessing_revision(project_id, cached=False) != \
```

Comment:
One question I have is around the explicit cached=False thing here. It seems slightly off for the caller to need to decide when is appropriate to use the cache or not. Does the cache not get properly invalidated when the revision is updated?

Reply:
Nevermind I see that it would be difficult to detect this update and invalidate the cache automatically

Reply:
@alex-hofsteede the idea here is just to detect changes that happened concurrently so it's quite isolated. Might make sense to rename this or provide a separate method.